types/office-js-preview/index.d.ts

```diff
@@ -22297,7 +22337,159 @@ declare namespace Excel {
+        /**
+        * An error caused by Excel 4.0 macros not being supported on the platform. Displays as error type #BLOCKED! in Excel.
+        *
+        * @remarks
+        * [Api set: ExcelApi BETA (PREVIEW ONLY)]
+        * @beta
+        */
+        xlmNotSupportedOnPlatform = "XlmNotSupportedOnPlatform",
+        /**
+        * An error caused by the image download coming from an untrusted URL. Displays as error type #BLOCKED! in Excel.
+        *
+        * @remarks
+        * [Api set: ExcelApi BETA (PREVIEW ONLY)]
+        * @beta
+        */
+        imageDownloadBlocked = "ImageDownloadBlocked",
+        /**
+        * An error caused by the image URL not being HTTPS. Displays as error type #BLOCKED! in Excel.
+        *
+        * @remarks
+        * [Api set: ExcelApi BETA (PREVIEW ONLY)]
+        * @beta
+        */
+        imageInvalidUrlNonHttps = "ImageInvalidUrlNonHttps",
+        /**
+        * An error caused by the image exceeding the size limit set in ExcelServerSettings.CacheBackedAsyncImageDownloaderMaxFileSizeInKBytes. Displays as error type #BLOCKED! in Excel.
+        *
+        * @remarks
+        * [Api set: ExcelApi BETA (PREVIEW ONLY)]
+        * @beta
+        */
+        imageExceedsSizeLimit = "ImageExceedsSizeLimit",
+        /**
+        * An error caused by the image exceeding the size limit for an entire workbook set in `ExcelWorkbookSizeMax`. Displays as error type #BLOCKED! in Excel.
+        *
+        * @remarks
+        * [Api set: ExcelApi BETA (PREVIEW ONLY)]
+        * @beta
+        */
+        imageExceedsTotalSizeLimit = "ImageExceedsTotalSizeLimit",
+        /**
+        * An error caused by the user exceeding their quota of concurrently active external code service execution environments. Displays as error type #BLOCKED! in Excel.
+        *
+        * @remarks
+        * [Api set: ExcelApi BETA (PREVIEW ONLY)]
+        * @beta
+        */
+        environmentExceedsQuota = "EnvironmentExceedsQuota",
+        /**
+        * An error caused by the user exceeding their daily quota of external code service execution environments. Displays as error type #BLOCKED! in Excel.
+        *
+        * @remarks
+        * [Api set: ExcelApi BETA (PREVIEW ONLY)]
+        * @beta
+        */
+        environmentExceedsQuotaDaily = "EnvironmentExceedsQuotaDaily",
+        /**
+        * An error caused by the user exceeding their monthly quota of external code service execution environments. Displays as error type #BLOCKED! in Excel.
+        *
+        * @remarks
+        * [Api set: ExcelApi BETA (PREVIEW ONLY)]
+        * @beta
+        */
+        environmentExceedsQuotaMonthly = "EnvironmentExceedsQuotaMonthly",
+        /**
+        * An error caused by the user who is not licensed for a feature which uses an external code service exceeding their monthly quota of external code service execution environments. Displays as error type #BLOCKED! in Excel.
```

Comment:
Thank you for going through these new type definitions so thoroughly! Really appreciate your effort. 

I'm not going to incorporate the contraction suggestions because, even though they're more aligned with our style guide, they're not grammatically incorrect and the overhead for making all the changes is pretty high (the process that generates these definitions in particular is challenging and outside the regular process). I did incorporate all your other suggestions!

Reply:
Heads up, I was able to include your contraction suggestions when the engineer made other changes in the source code! They'll make it to DefinitelyTyped eventually.